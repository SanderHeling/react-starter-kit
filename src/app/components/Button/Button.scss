.Button {
    display: inline-flex;

    padding: 10px 20px;

    cursor: pointer;
    transition: background 300ms;

    color: #fff;
    background: palevioletred;

    &:focus {
        outline: none;
    }

    &:hover {
        background: darken(palevioletred, 5%);
    }
}
