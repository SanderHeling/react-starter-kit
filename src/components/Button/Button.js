import React from 'react';
import PropTypes from 'prop-types';
import cs from 'classtr';

import styles from './Button.scss';

const Button = ({ className, ...other }) => (
    <div
        className={cs([styles.Button, className])}
        role="button"
        tabIndex={0}
        {...other}
    />
);

Button.propTypes = {
    className: PropTypes.string,
};

Button.defaultProps = {
    className: '',
};

export default Button;
