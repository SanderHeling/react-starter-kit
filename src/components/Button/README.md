## Examples

```jsx
<Button>Button</Button>
```
