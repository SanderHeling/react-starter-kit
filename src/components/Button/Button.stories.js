import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';

import Button from './Button';

// Add padding to story
const PaddedDecorator = storyFn => (
    <div style={{ padding: '20px' }}>{storyFn()}</div>
);

storiesOf('Button', module)
    .addDecorator(PaddedDecorator)
    .add(
        'with content',
        withInfo()(() => <Button onClick={action('clicked')}>Button</Button>)
    )
    .add(
        'with custom class',
        withInfo()(() => (
            <Button onClick={action('clicked')} className="customClass">
                Button
            </Button>
        ))
    );
