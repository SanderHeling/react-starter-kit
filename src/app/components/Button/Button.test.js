import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

test('should render correctly', () => {
    const component = renderer.create(<Button>Text</Button>).toJSON();
    expect(component).toMatchSnapshot();
});
