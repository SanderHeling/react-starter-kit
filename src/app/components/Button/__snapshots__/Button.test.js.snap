// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="Button "
  role="button"
  tabIndex={0}
>
  Text
</div>
`;
